cmd/cstor-volume-mgmt/controller/volume-controller/after_queue_handler.go

```diff
@@ -0,0 +1,160 @@
+	}
+	_, err = c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// cStorVolumeEventHandler is to handle cstor volume related events.
+func (c *CStorVolumeController) cStorVolumeEventHandler(operation common.QueueOperation, cStorVolumeGot *apis.CStorVolume) (common.CStorVolumeStatus, error) {
+	volume.RunnerVar = util.RealRunner{}
+	volume.FileOperatorVar = util.RealFileOperator{}
+	volume.UnixSockVar = util.RealUnixSock{}
+	glog.Infof("%v event received for volume : %v ", operation, cStorVolumeGot.Spec.VolumeName)
+	switch operation {
+	case common.QOpAdd:
+		// CheckValidVolume is to check if volume attributes are correct.
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusOffline, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusFailed, err
+		}
+		break
+
+	case common.QOpModify:
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusInvalid, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusFailed, err
+		}
+		break
+
+	case common.QOpDestroy:
+		return common.CVStatusIgnore, nil
+	}
+
+	return common.CVStatusIgnore, nil
+}
+
+// enqueueCstorVolume takes a CStorVolume resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than CStorVolumes.
+func (c *CStorVolumeController) enqueueCStorVolume(obj *apis.CStorVolume, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.Key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// getVolumeResource returns object corresponding to the resource key
+func (c *CStorVolumeController) getVolumeResource(key string) (*apis.CStorVolume, error) {
+	// Convert the key(namespace/name) string into a distinct name
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s, err: %v", key, err))
+		return nil, nil
+	}
+
+	cStorVolumeGot, err := c.clientset.OpenebsV1alpha1().CStorVolumes().Get(name, metav1.GetOptions{})
+	if err != nil {
+		// The cStorVolume resource may no longer exist, in which case we stop
+		// processing.
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("cStorVolumeGot '%s' in work queue no longer exists", key))
+			return nil, nil
+		}
+
+		return nil, err
+	}
+	return cStorVolumeGot, nil
+}
+
+// IsValidCStorVolumeMgmt is to check if the volume request is for particular pod/application.
+func IsValidCStorVolumeMgmt(cStorVolume *apis.CStorVolume) bool {
+	if os.Getenv("OPENEBS_IO_CSTOR_VOLUME_ID") == string(cStorVolume.ObjectMeta.UID) {
+		glog.V(2).Infof("right sidecar for the cstor volume with id : %s", cStorVolume.ObjectMeta.UID)
```

Comment:
it would be better not to use the term sidecar in logs. Right CStorVolume or any other can be used. Since sidecar is design pattern and not advisable to use in log.